### PR TITLE
feat: simplify queue batching options

### PR DESCRIPTION
**Pull request by saikumarrs**
## PR Description

Simplified the batch options.

## Linear task (optional)

Linear task link

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: packages/analytics-js-plugins/src/utilities/retryQueue/RetryQueue.ts [hunk 1]
```diff
@@ -138,27 +138,15 @@ class RetryQueue implements IQueue<QueueItemData> {
...
-    }
-
-    // Set upper cap on the batch size
     if (this.batch.enabled) {
-      let maxSize = this.batch?.maxSize ?? DEFAULT_MAX_BATCH_SIZE_BYTES;
-      if (maxSize > DEFAULT_MAX_BATCH_SIZE_BYTES) {
-        maxSize = DEFAULT_MAX_BATCH_SIZE_BYTES;
+      this.batch.maxSize = batchOptions.maxSize || DEFAULT_MAX_BATCH_SIZE_BYTES;
+      // Set upper cap on the batch size
+      if (this.batch.maxSize > DEFAULT_MAX_BATCH_SIZE_BYTES) {
```
Comment (bardisg) [inline packages/analytics-js-plugins/src/utilities/retryQueue/RetryQueue.ts:144]: you can use Math.min(batchOptions.maxSize || DEFAULT_MAX_BATCH_SIZE_BYTES, DEFAULT_MAX_BATCH_SIZE_BYTES)

File: packages/analytics-js-plugins/__tests__/utilities/retryQueue/RetryQueue.test.ts [hunk 2]
```diff
@@ -909,7 +909,8 @@ describe('Queue', () => {
     expect(batchQueue.batch).toEqual({
       enabled: true,
       maxSize: 1024,
-      maxItems: 1,
+      maxItems: '1',
```
Comment (bardisg) [inline packages/analytics-js-plugins/__tests__/utilities/retryQueue/RetryQueue.test.ts:912]: I am confused, are these strings instead of numbers?

Comment (saikumarrs) [reply packages/analytics-js-plugins/__tests__/utilities/retryQueue/RetryQueue.test.ts:912]: Yeah, I removed the data type coercion. The user has to pass sane values to the options.

File: packages/analytics-js-plugins/src/utilities/retryQueue/RetryQueue.ts (hunk 1 above)
Comment (saikumarrs) [reply packages/analytics-js-plugins/src/utilities/retryQueue/RetryQueue.ts]: Thanks. Done.

**Review by bardisg — APPROVED** [review]

**Pull request merged**